Add 'My Tales' filter back to Browse view

bodom0015 opened the pull request:
### Problem
After refactoring the Browse view, the "My Tales" filter has been disabled.

Fixes #511 

### Approach
Re-enable the "My Tales" filter.

NOTE: This PR does not alter the method by which the tales are searched or filtered.

### How to Test
Prerequisites: At least one Tale accessible not owned by you

1. Checkout and run this branch locally, rebuild the dashboard
2. Login to the WholeTale Dashboard
3. Navigate to the Browse view
    * By default, you should be shown "All Tales" (API automatically filters GET /tales to only those accessible to the current user)
4. On the top-left, click the "My Tales" filter button
    * You should see the contents of the view change to only show you items you have created
    * NOTE: this could be adjusted to filter on `_accessLevel`, if desired

Xarthisius commented:
LGTM! Minor nitpick: when user has no Tales text in "My Tales" tab says: `There are no tales to display for applied filter: all`. The following patch fixed it for me:

```
diff --git a/app/components/ui/tale-browser/component.js b/app/components/ui/tale-browser/component.js
index 0fb8ec7..058465e 100644
--- a/app/components/ui/tale-browser/component.js
+++ b/app/components/ui/tale-browser/component.js
@@ -8,7 +8,6 @@ import $ from 'jquery';
 const O = EmberObject.create.bind(EmberObject);
 
 export default Component.extend({
-  currentTab: 'all',
   store: service(),
   userAuth: service('user-auth'),
   apiCall: service('api-call'),
diff --git a/app/components/ui/tale-browser/template.hbs b/app/components/ui/tale-browser/template.hbs
index f7f60e7..f2a126d 100644
--- a/app/components/ui/tale-browser/template.hbs
+++ b/app/components/ui/tale-browser/template.hbs
@@ -149,7 +149,7 @@
     
     {{#unless modelsInView.length}}
         <div class="ui center aligned segment" style="color: black;">
-            There are no tales to display for applied filter: {{currentTab}}
+            There are no tales to display for applied filter: {{filter}}
         </div>
     {{/unless}}
     {{#if loadingTales}}

```

Xarthisius commented:
Noticed something more serious: deleting a Tale in "My Tales" tab renders dashboard somewhat unresponsive if you go back to the "All Tales" (e.g. there's no view button when I hover over a Tale).

I see the following backtrace in the console:

```app.js:22 TypeError: Cannot read property 'name' of undefined
    at Class.<anonymous> (component.js:13)
    at ComputedPropertyPrototype.get (ember-metal.js:4069)
    at get (ember-metal.js:3439)
    at RootPropertyReference.compute (ember-glimmer.js:326)
    at RootPropertyReference.value (ember-glimmer.js:203)
    at valueOf (runtime.js:4236)
    at Array.map (<anonymous>)
    at CapturedPositionalArguments.value (runtime.js:4216)
    at SimpleHelperReference.compute (ember-glimmer.js:506)
    at SimpleHelperReference.value (ember-glimmer.js:203)
```
`component.js` from above stack is `app/components/ui/delete-modal/component.js`

### Steps to reproduce
1. Create Ligo Tale.
2. Copy and Launch Ligo
3. Stop "Copy" Ligo Tale
4. Navigate to Browse > My Tales and delete copied Ligo Tale
5. (optionally) Navigate to Browse > All Tales and hover over Ligo Tale

bodom0015 commented:
Thanks, @Xarthisius!

I think I have addressed these issues.. I'm not sure how suddenly the `delete-modal` was acting up, but I've adjusted it to use only the modelName (since this could either be `name` or `title`, better to just let the caller decide which string to pass in).

This appears to have gotten rid of that gross stack trace, and the "View" buttons appear to stay now.

Let me know if this clears up the issue for you as well!

Xarthisius commented:
@bodom0015 after updating to 723700c8c6165b585b76e43115c8f1a69ab9e3ab and login I get:

![bug](https://user-images.githubusercontent.com/352673/61734168-fcaca680-ad46-11e9-9574-6d58bfa4a096.png)

here's the console output:

```
Attributes updated
wholetale-c9f8dd9c9489daf30eddc6445e66f441.js:38 TypeError: Cannot read property 'syscall' of null
    at e.evaluate (vendor-468b71615cf72303033dfaa7aa6e61b3.js:1146)
    at e.evaluateSyscall (vendor-468b71615cf72303033dfaa7aa6e61b3.js:1387)
    at e.evaluateInner (vendor-468b71615cf72303033dfaa7aa6e61b3.js:1381)
    at e.evaluateOuter (vendor-468b71615cf72303033dfaa7aa6e61b3.js:1381)
    at e.next (vendor-468b71615cf72303033dfaa7aa6e61b3.js:1518)
    at e.next (vendor-468b71615cf72303033dfaa7aa6e61b3.js:1519)
    at e.render (vendor-468b71615cf72303033dfaa7aa6e61b3.js:2081)
    at e.runInTransaction (vendor-468b71615cf72303033dfaa7aa6e61b3.js:2289)
    at t.e._renderRoots (vendor-468b71615cf72303033dfaa7aa6e61b3.js:2104)
    at t.e._renderRootsTransaction (vendor-468b71615cf72303033dfaa7aa6e61b3.js:2109)
```

bodom0015 commented:
@Xarthisius very weird... that stack trace is definitely not from our code (`vendor.js`), and we shouldn't be using `syscall` directly anywhere since I'm not even sure what that is or does...

Have you tried in incognito? As much as I hate the "it works on my machine" argument, I'm not sure what that error means or where it could have come from

Xarthisius commented:
> @Xarthisius very weird... that stack trace is definitely not from our code (`vendor.js`), and we shouldn't be using `syscall` directly anywhere since I'm not even sure what that is or does...
> 
> Have you tried in incognito? As much as I hate the "it works on my machine" argument, I'm not sure what that error means or where it could have come from

I nuked `node_modules` and `bower_components`, installed everything again, tried incognito in chrome and even installed firefox. Same behavior.

Xarthisius commented:
@bodom0015 removing
```
{{ui/copy-on-launch-modal taleToCopy=taleToCopy modelsInView=modelsInView}}
```
from `app/components/ui/tale-browser/template.hbs` "fixes" it.

Xarthisius commented on app/components/ui/tale-browser/template.hbs:
```diff
@@ -424,25 +423,4 @@
-    </div>
-  </div>
-</div>
+{{ui/copy-on-launch-modal taleToCopy=taleToCopy modelsInView=modelsInView}}
```
/me waves at @bodom0015

Xarthisius commented:
> @bodom0015 removing
> 
> ```
> {{ui/copy-on-launch-modal taleToCopy=taleToCopy modelsInView=modelsInView}}
> ```
> 
> from `app/components/ui/tale-browser/template.hbs` "fixes" it.

Resolved by merging https://github.com/whole-tale/dashboard/pull/516
